### PR TITLE
build(mk): update Envoy to v1.25.3, use new kumahq/envoy-builds releases

### LINKED ISSUE #4619: Update to Envoy 1.25
### Description

Need to move to envoy 1.23 now that it's release

### PR DESCRIPTION
**Pull request by michaelbeaumont**
@lahabana could you take a look at how to merge this with the latest `mk/envoy.mk` changes?

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues -- closes #4619
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Review by lahabana — APPROVED** [review]
I've checked that everything builds and that envoy builds on darwin work correctly.

I'm excited to see this finally land!

File: test/framework/universal_app.go [hunk 1]
```diff
@@ -436,7 +436,7 @@ func (s *UniversalApp) CreateDP(
 		"/usr/bin/kuma-dp", "run",
 		"--cp-address=" + cpAddress,
 		"--dataplane-token-file=/kuma/token-" + name,
-		"--binary-path", "/usr/local/bin/envoy",
+		"--binary-path", "/usr/bin/envoy",
```
Comment (lahabana) [inline test/framework/universal_app.go:439]: Should we just not set this? I feel like this should work out of the box

**Comment by lukidzi** [comment]
I think we need to configure  https://github.com/kumahq/kuma/blob/master/pkg/hds/tracker/healthcheck_generator.go#L92 to override source address to 127.0.0.6 and later we can enable this test https://github.com/kumahq/kuma/blob/master/test/e2e_env/multizone/healthcheck/healthcheck.go#L32

**Comment by michaelbeaumont** [comment]
> I think we need to configure https://github.com/kumahq/kuma/blob/master/pkg/hds/tracker/healthcheck_generator.go#L92 to override source address to `127.0.0.6`(like [here](https://github.com/kumahq/kuma/blob/master/pkg/xds/generator/inbound_proxy_generator.go#L46)) and later we can enable this test https://github.com/kumahq/kuma/blob/master/test/e2e_env/multizone/healthcheck/healthcheck.go#L32

Is this necessary for everything to work properly or is it another PR to finish https://github.com/kumahq/kuma/issues/4772?

**Comment by lukidzi** [comment]
> > I think we need to configure https://github.com/kumahq/kuma/blob/master/pkg/hds/tracker/healthcheck_generator.go#L92 to override source address to `127.0.0.6`(like [here](https://github.com/kumahq/kuma/blob/master/pkg/xds/generator/inbound_proxy_generator.go#L46)) and later we can enable this test https://github.com/kumahq/kuma/blob/master/test/e2e_env/multizone/healthcheck/healthcheck.go#L32
> 
> Is this necessary for everything to work properly or is it another PR to finish #4772?

You are right. Forgot about this 👍

**Review by lukidzi — APPROVED** [review]

File: UPGRADE.md [hunk 2]
```diff
@@ -21,23 +27,23 @@ List of unsupported configuration options:
...
 
-### Gateway API
+#### Gateway API
 
 We now support version `v0.6.0` of the Gateway API. See the [upstream API
 changes](https://github.com/kubernetes-sigs/gateway-api/releases/tag/v0.6.0) for
 more info.
 
-### Longer name of the resource in postgres
+#### Longer name of the resource in postgres
```
Comment (lahabana) [inline UPGRADE.md:46]: That's not k8s.
```suggestion
### Universal
#### Longer name of the resource in postgres
```

**Review by lahabana — APPROVED** [review]

**Pull request merged**